run_tests.py

```diff
@@ -80,24 +83,47 @@ class GRREverythingTestLoader(test_lib.GRRTestLoader):
   base_class = test_lib.GRRBaseTest
 
 
+class GRROneTestLoader(test_lib.GRRTestLoader):
+  """Load one specific GRR test case."""
+  base_class = test_lib.GRRBaseTest
+
+  def loadTestsFromModule(self, _):
+    """Just return all the tests as if they were in the same module."""
+    test_cases = [
+        self.loadTestsFromTestCase(x) for x in self.base_class.classes.values()
+        if x.__name__ == self.test_class_name]
+
+    return self.suiteClass(test_cases)
+
+  def __init__(self, test_class_name):
+    self.test_class_name = test_class_name
+
+    super(GRROneTestLoader, self).__init__()
+
+
 def RunTest(test_suite, stream=None):
   out_fd = stream
   if stream:
     out_fd = StringIO.StringIO()
 
   try:
-    test_lib.GrrTestProgram(argv=[sys.argv[0], test_suite],
-                            testLoader=GRREverythingTestLoader(
-                                labels=flags.FLAGS.labels),
-                            testRunner=unittest.TextTestRunner(
-                                stream=out_fd))
+    program = test_lib.GrrTestProgram(
+                argv=[sys.argv[0], test_suite],
+                testLoader=GRROneTestLoader(
+                    test_class_name=test_suite),
+                testRunner=unittest.TextTestRunner(
+                    stream=out_fd))
+  except SystemExit as e:
+    # must include this section, as GrrTestProgram issues
```

Comment:
Maybe that's the problem - they need to run in their own processes even if there is only one process. This way the test suite can be aware if a test crashes for example (otherwise it brings the entire test suite down with it). I can see that it uses a different code path when processes == 1 - maybe we should change that.

The failure you are referring to shows that the test is killed probably due to OOM. We see this occasionally in Rekall too - it seems that travis does not have a lot of memory for tests and its easy to exhaust it. Maybe we should either rerun the test if its killed or consider it to pass.

The whole idea with the test suite is that it will run all the tests as stand alone programs - this allows us to run anything including c++ tests the same way. The test is not called from a function - it is called from a runner which is forked for it.

BTW I have no idea what you mean by Google flags? We use argparse which is a standard python module.